handler/src/main/java/io/netty/handler/ssl/AbstractSniHandler.java

```diff
@@ -57,156 +54,152 @@ protected void decode(ChannelHandlerContext ctx, ByteBuf in, List<Object> out) t
-                        case SslUtils.SSL_CONTENT_TYPE_CHANGE_CIPHER_SPEC:
-                        case SslUtils.SSL_CONTENT_TYPE_ALERT:
-                            final int len = SslUtils.getEncryptedPacketLength(in, readerIndex);
+                // tls, but not handshake command
+                switchStatement:
+                switch (command) {
+                case SslUtils.SSL_CONTENT_TYPE_CHANGE_CIPHER_SPEC:
+                case SslUtils.SSL_CONTENT_TYPE_ALERT:
+                    final int len = SslUtils.getEncryptedPacketLength(in, readerIndex);
 
-                            // Not an SSL/TLS packet
-                            if (len == SslUtils.NOT_ENCRYPTED) {
-                                handshakeFailed = true;
-                                NotSslRecordException e = new NotSslRecordException(
-                                        "not an SSL/TLS record: " + ByteBufUtil.hexDump(in));
-                                in.skipBytes(in.readableBytes());
+                    // Not an SSL/TLS packet
+                    if (len == SslUtils.NOT_ENCRYPTED) {
+                        handshakeFailed = true;
+                        NotSslRecordException e = new NotSslRecordException(
+                                "not an SSL/TLS record: " + ByteBufUtil.hexDump(in));
+                        in.skipBytes(in.readableBytes());
 
-                                SslUtils.notifyHandshakeFailure(ctx, e, true);
-                                throw e;
-                            }
-                            if (len == SslUtils.NOT_ENOUGH_DATA ||
-                                    writerIndex - readerIndex - SslUtils.SSL_RECORD_HEADER_LENGTH < len) {
-                                // Not enough data
-                                return;
-                            }
-                            // increase readerIndex and try again.
-                            in.skipBytes(len);
-                            continue;
-                        case SslUtils.SSL_CONTENT_TYPE_HANDSHAKE:
-                            final int majorVersion = in.getUnsignedByte(readerIndex + 1);
-
-                            // SSLv3 or TLS
-                            if (majorVersion == 3) {
-                                final int packetLength = in.getUnsignedShort(readerIndex + 3) +
-                                                         SslUtils.SSL_RECORD_HEADER_LENGTH;
-
-                                if (readableBytes < packetLength) {
-                                    // client hello incomplete; try again to decode once more data is ready.
-                                    return;
-                                }
+                        SslUtils.notifyHandshakeFailure(ctx, e, true);
+                        throw e;
+                    }
+                    if (len == SslUtils.NOT_ENOUGH_DATA ||
+                        writerIndex - readerIndex - SslUtils.SSL_RECORD_HEADER_LENGTH < len) {
```

Comment:
should this be `in.readableBytes()` ?

Reply:
What's "this" in this comment? `len`?

Reply:
Maybe @carl-mastrangelo is talking about `writerIndex - readerIndex` ... which maybe now we can use `readableBytes`?

Reply:
@carl-mastrangelo - Ping

Reply:
I just found out this tab was still open.  yes that is what I meant.

Reply:
thanks @carl-mastrangelo ... @Lukasa ^^